modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java

```diff
@@ -292,6 +337,153 @@ public void testRollbackOnInvalidRequest() throws IOException {
+    public void testCreateServiceProviderRequestWithInvalidIdTokenEncryptedResponseEnc() throws IOException {
+        HttpPost request = new HttpPost(getPath());
+        setRequestHeaders(request);
+
+        JSONArray grantTypes = new JSONArray();
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_AUTHORIZATION_CODE);
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_IMPLICIT);
+
+        JSONArray redirectURI = new JSONArray();
+        redirectURI.add(OAuthDCRMConstants.REDIRECT_URI);
+
+        JSONObject dcrRequest = new JSONObject();
+        dcrRequest.put(OAuthDCRMConstants.CLIENT_NAME, DUMMY_DCR_APP);
+        dcrRequest.put(OAuthDCRMConstants.GRANT_TYPES, grantTypes);
+        dcrRequest.put(OAuthDCRMConstants.REDIRECT_URIS, redirectURI);
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ALG,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ALG_VALUE);
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC,
+                OAuthDCRMConstants.INVALID_ID_TOKEN_ENCRYPTED_RESPONSE_ENC_VALUE);
+
+
+        StringEntity entity = new StringEntity(dcrRequest.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 500, "Service Provider " +
+                "creation request with invalid id_token_encrypted_response_enc should have returned an internal " +
+                "server error");
+
+        EntityUtils.consume(response.getEntity());
+
+    }
+
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 10, description = "Try to register service provider with " +
+            "id_token_encrypted_response_enc without id_token_encrypted_response_alg")
+    public void testCreateServiceProviderRequestWithoutIdTokenEncryptedResponseAlg() throws IOException {
+        HttpPost request = new HttpPost(getPath());
+        setRequestHeaders(request);
+
+        JSONArray grantTypes = new JSONArray();
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_AUTHORIZATION_CODE);
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_IMPLICIT);
+
+        JSONArray redirectURI = new JSONArray();
+        redirectURI.add(OAuthDCRMConstants.REDIRECT_URI);
+
+        JSONObject dcrRequest = new JSONObject();
+        dcrRequest.put(OAuthDCRMConstants.CLIENT_NAME, DUMMY_DCR_APP);
+        dcrRequest.put(OAuthDCRMConstants.GRANT_TYPES, grantTypes);
+        dcrRequest.put(OAuthDCRMConstants.REDIRECT_URIS, redirectURI);
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC_VALUE);
+
+
+        StringEntity entity = new StringEntity(dcrRequest.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 400, "Service Provider " +
```

Comment:
Is this behaviour mandarated by the spec?
Or can the OP implementation decide this?

Reply:
This is mandated by the spec 
[https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata)